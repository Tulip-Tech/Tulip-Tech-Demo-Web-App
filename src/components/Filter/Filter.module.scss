@use 'src/styles/variables';
@use 'src/styles/theme';
@use 'src/styles/mixins/responsive';

.filterRow {
  display: flex;
  align-items: center;
  margin-left: 16px;

  > button {
    margin: 0 4px;
  }
}

.dropDown {
  margin-bottom: 0;
}

.filterDropDown {
  display: flex;
  align-items: flex-end;
}
