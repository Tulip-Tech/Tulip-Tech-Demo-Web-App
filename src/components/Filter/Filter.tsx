import React, { Fragment, FC } from 'react';
import { useTranslation } from 'react-i18next';

import styles from './Filter.module.scss';

import Dropdown from '#components/Dropdown/Dropdown';
import Button from '#components/Button/Button';
import useBreakpoint, { Breakpoint } from '#src/hooks/useBreakpoint';

type Props = {
  name: string;
  value: string;
  valuePrefix?: string;
  defaultLabel: string;
  options: string[];
  forceDropdown?: boolean;
  setValue: (value: string) => void;
};

const Filter: FC<Props> = ({ name, value, defaultLabel, options, setValue, valuePrefix = '', forceDropdown = false }) => {
  const { t } = useTranslation('common');
  const breakpoint: Breakpoint = useBreakpoint();

  if (!options.length) {
    return null;
  }
  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => setValue(event.target.value);

  const showFilterRow = !forceDropdown && breakpoint >= Breakpoint.md && options.length < 6;

  return (
    <Fragment>
      {showFilterRow ? (
        <div className={styles.filterRow} role="listbox" aria-label={t('filter_videos_by', { name })}>
          {options.map((option) => (
            <Button label={`${valuePrefix}${option}`} onClick={() => setValue(option)} key={option} active={value === option} role="option" />
          ))}
          <Button label={defaultLabel} onClick={() => setValue('')} active={value === ''} key={defaultLabel} role="option" />
        </div>
      ) : (
        <div className={styles.filterDropDown}>
          <Dropdown
            className={styles.dropDown}
            size="small"
            options={options}
            defaultLabel={defaultLabel}
            valuePrefix={valuePrefix}
            name={name}
            value={value}
            onChange={handleChange}
            aria-label={t('filter_videos_by', { name })}
          />
        </div>
      )}
    </Fragment>
  );
};

export default Filter;
