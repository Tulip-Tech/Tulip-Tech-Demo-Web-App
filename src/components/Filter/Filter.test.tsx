import * as React from 'react';
import { render } from '@testing-library/react';

import Filter from './Filter';

const options = ['x', 'y', 'z'];

describe('<Filter>', () => {
  it('renders Filter', () => {
    const { container } = render(<Filter name="categories" value="aa" defaultLabel="bb" options={options} setValue={(event) => event} />);
    expect(container).toMatchSnapshot();
  });
});
