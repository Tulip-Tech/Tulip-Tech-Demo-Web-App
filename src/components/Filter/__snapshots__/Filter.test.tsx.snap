// Vitest Snapshot v1

exports[`<Filter> > renders Filter 1`] = `
<div>
  <div
    aria-label="filter_videos_by"
    class="_filterRow_2153bc"
    role="listbox"
  >
    <button
      class="_button_f8f296 _default_f8f296 _outlined_f8f296"
      role="option"
    >
      <span>
        x
      </span>
    </button>
    <button
      class="_button_f8f296 _default_f8f296 _outlined_f8f296"
      role="option"
    >
      <span>
        y
      </span>
    </button>
    <button
      class="_button_f8f296 _default_f8f296 _outlined_f8f296"
      role="option"
    >
      <span>
        z
      </span>
    </button>
    <button
      class="_button_f8f296 _default_f8f296 _outlined_f8f296"
      role="option"
    >
      <span>
        bb
      </span>
    </button>
  </div>
</div>
`;
